vm/src/deserializer.cpp

```diff
@@ -0,0 +1,42 @@
+#include "deserializer.h"
+#include <cassert>
+#include <stdexcept>
+
+namespace Borges {
+
+Deserializer::Deserializer(std::size_t n, const double* p)
+    : size_(n), data_(p), index_(0)
+{
+    assert(data_);
+}
+
+void
+Deserializer::deserialize(double* p)
+{
+    if (index_ == size_) {
+        throw std::runtime_error("buffer underflow");
+    }
+    double x = data_[index_];
+    ++index_;
+    *p = x;
+}
+
+void
+deserialize(Deserializer& d, bool* p)
+{
+    *p = deserialize<double>(d);
+}
+
+void
+deserialize(Deserializer& d, double* p)
```

Comment:
why does a void function use the return statement here ?

Reply:
I think that's a leftover from an earlier iteration when Deserializer:deserialize still returned something. I agree that it should be removed.